repair/repair.cc

```diff
@@ -1400,29 +1425,6 @@ static int do_repair_start(seastar::sharded<database>& db, sstring keyspace,
         ranges = std::move(intersections);
     }
 
-    std::vector<sstring> cfs;
-    std::vector<schema_ptr> schemas;
-    if (options.column_families.size()) {
-        cfs = options.column_families;
-        for (auto& cf : cfs) {
-            try {
-                schemas.push_back(db.local().find_column_family(keyspace, cf).schema());
-            } catch(...) {
-                throw std::runtime_error(format("No column family '{}' in keyspace '{}'", cf, keyspace));
-            }
-        }
-    } else {
-        std::tie(cfs, schemas) = list_column_families(db.local(), keyspace);
-    }
-    if (cfs.empty()) return id;
-    for (size_t idx = 1; idx < schemas.size(); ++idx) {
-        if (schemas[idx - 1]->get_partitioner() != schemas[idx]->get_partitioner()) {
-            throw std::runtime_error(
-                    format("All column families repaired together have to have the same partitioner. Different partitioners found: {} and {}",
-                        schemas[idx - 1]->get_partitioner(), schemas[idx]->get_partitioner()));
-        }
-    }
-    dht::i_partitioner& partitioner = schemas[0]->get_partitioner();
```

Comment:
Why do you want to move the code up? Even if there is a reason to do so, please split it out to a separate move-only patch. It is much easier to review.

Reply:
This change is gone once I moved the check to repair_info constructor.